compiler/plugins/target/AMD-AIE/aievec/test/canonicalize_transfer_write_for_load.mlir

```diff
@@ -0,0 +1,158 @@
+// CHECK-DAG: %[[C0:.*]] = arith.constant 0 : index
+// CHECK-DAG: %[[C32:.*]] = arith.constant 32 : index
+// CHECK-DAG: %[[C224:.*]] = arith.constant 224 : index
+// CHECK-DAG: %[[CST_31:.*]] = arith.constant dense<0> : vector<31xi32>
+// CHECK-DAG: %[[CST_32:.*]] = arith.constant dense<0> : vector<32xi32>
+// CHECK-DAG: %[[ALLOC:.*]] = memref.alloc() : memref<255xi32>
+// CHECK: scf.for %arg0 = %[[C0]] to %[[C224]] step %[[C32]] {
+// CHECK-NEXT: vector.transfer_write %[[CST_32]], %[[ALLOC]][%arg0] {in_bounds = [true]} : vector<32xi32>, memref<255xi32>
+// CHECK-NEXT: }
+// CHECK-NEXT: vector.transfer_write %[[CST_31]], %[[ALLOC]][%[[C224]]] {in_bounds = [true]} : vector<31xi32>, memref<255xi32>
+// CHECK-NEXT: return
+
+
+// -----
+// When the transfer_write is smaller than than the target device store size (128)
+// the IR is left unchanged.
+#executable_target_ = #hal.executable.target<"", "", {target_device = "npu1_4col"}>
+module attributes {hal.executable.target = #executable_target_} {
+  func.func @test_small_non_divisible() {
+    %c0 = arith.constant 0 : index
+    %alloc = memref.alloc() : memref<5xi32>
+    %cst = arith.constant dense<0> : vector<5xi32>
+    vector.transfer_write %cst, %alloc[%c0] {in_bounds = [true]} : vector<5xi32>, memref<5xi32>
+    return
+  }
+}
+// CHECK-LABEL: func.func @test_small_non_divisible()
+// CHECK-DAG: %[[C0:.*]] = arith.constant 0 : index
+// CHECK-DAG: %[[ALLOC:.*]] = memref.alloc() : memref<5xi32>
+// CHECK-DAG: %[[CST:.*]] = arith.constant dense<0> : vector<5xi32>
+// CHECK: vector.transfer_write %[[CST]], %[[ALLOC]][%[[C0]]] {in_bounds = [true]} : vector<5xi32>, memref<5xi32>
+// CHECK: return
+
+
+// -----
+// In this test we write 100xi32 elements, this 400 byte write is decomposed
+// 3x128 bytes then 1x16 bytes (3x32 elements, then 1x4 elements). The write
+// is also offset into the memref, and the element written in 1 not 0.
+#executable_target_ = #hal.executable.target<"", "", {target_device = "npu4"}>
+module attributes {hal.executable.target = #executable_target_} {
+  func.func @test_non_zero_and_offset() {
+    %c7 = arith.constant 7 : index
+    %alloc = memref.alloc() : memref<256xi32>
+    %cst = arith.constant dense<1> : vector<100xi32>
+    vector.transfer_write %cst, %alloc[%c7] {in_bounds = [true]} : vector<100xi32>, memref<256xi32>
+    return
+  }
+}
+// CHECK-LABEL: func.func @test_non_zero_and_offset()
+// CHECK-DAG: %[[C7:.*]] = arith.constant 7 : index
+// CHECK-DAG: %[[C32:.*]] = arith.constant 32 : index
+// CHECK-DAG: %[[C103:.*]] = arith.constant 103 : index
+// CHECK-DAG: %[[CST_4:.*]] = arith.constant dense<1> : vector<4xi32>
+// CHECK-DAG: %[[CST_32:.*]] = arith.constant dense<1> : vector<32xi32>
+// CHECK-DAG: %[[ALLOC:.*]] = memref.alloc() : memref<256xi32>
+// CHECK: scf.for %arg0 = %[[C7]] to %[[C103]] step %[[C32]] {
+// CHECK-NEXT: vector.transfer_write %[[CST_32]], %[[ALLOC]][%arg0] {in_bounds = [true]} : vector<32xi32>, memref<256xi32>
+// CHECK-NEXT: }
```

Comment:
For this and other lit tests could you align/indent them? Like the following :-
```suggestion
// CHECK-DAG:  %[[ALLOC:.*]] = memref.alloc() : memref<256xi32>
// CHECK:      scf.for %arg0 = %[[C7]] to %[[C103]] step %[[C32]] {
// CHECK-NEXT:     vector.transfer_write %[[CST_32]], %[[ALLOC]][%arg0] {in_bounds = [true]} : vector<32xi32>, memref<256xi32>
// CHECK-NEXT: }
```
Helps in skimming through better.